src/modules/editingTask.js

```diff
@@ -0,0 +1,44 @@
+import modifyTask from './modifyTask.js';
+
+function finishEditing(event) {
+  const selectedElementClass = `.${event.currentTarget.parentNode.parentNode.classList[0]}`;
+  const elementLi = document.querySelector(selectedElementClass);
+  const elementInput = event.currentTarget;
+  elementLi.classList.remove('editing-background');
+  elementInput.classList.remove('editing-background');
+  elementInput.addEventListener('input', modifyTask(elementInput.value, selectedElementClass));
+  const icon = document.querySelector(`${selectedElementClass} > img`);
+  icon.setAttribute('src', './assets/images/dot.png');
+}
+
+function editingTask(event) {
+  document.querySelectorAll('.editing-background').forEach((element) => {
+    element.classList.remove('editing-background');
+  });
+  const elementLi = event.currentTarget;
+  const elementInput = event.currentTarget.firstElementChild.lastElementChild;
+  elementInput.addEventListener('blur', finishEditing);
+  elementLi.classList.add('editing-background');
+  elementInput.classList.add('editing-background');
+  const icon = document.querySelector('.editing-background > img');
+  icon.setAttribute('src', './assets/images/trash-can.png');
+  const p = event.currentTarget;
+  icon.addEventListener('click', () => {
+    const list = JSON.parse(localStorage.getItem('Tasks'));
+    const newlist = [];
+    list.forEach((element) => {
+      if (element.index !== Number(p.classList[0].slice(5))) {
+        newlist.push(element);
+      }
+    });
+    let newindex = 0;
+    elementLi.remove();
+    newlist.forEach((element) => {
+      element.index = newindex;
+      newindex += 1;
```

Comment:
- [OPTIONAL] Nice work done on your Todo list, kindly set the property index to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5). It is currently starting at 0.